internal/libyaml/yamlh.go

```diff
@@ -0,0 +1,804 @@
+	DEFAULT_MAPPING_TAG  = MAP_TAG // The default mapping tag is !!map.
+)
+
+type NodeType int
+
+// Node types.
+const (
+	// An empty node.
+	NO_NODE NodeType = iota
+
+	SCALAR_NODE   // A scalar node.
+	SEQUENCE_NODE // A sequence node.
+	MAPPING_NODE  // A mapping node.
+)
+
+// An element of a sequence node.
+type NodeItem int
+
+// An element of a mapping node.
+type NodePair struct {
+	key   int // The key of the element.
+	value int // The value of the element.
+}
+
+// The node structure.
+type Node struct {
+	typ NodeType // The node type.
+	tag []byte   // The node tag.
+
+	// The node data.
+
+	// The scalar parameters (for SCALAR_NODE).
+	scalar struct {
+		value  []byte      // The scalar value.
+		length int         // The length of the scalar value.
+		style  ScalarStyle // The scalar style.
+	}
+
+	// The sequence parameters (for YAML_SEQUENCE_NODE).
+	sequence struct {
+		items_data []NodeItem    // The stack of sequence items.
+		style      SequenceStyle // The sequence style.
+	}
+
+	// The mapping parameters (for MAPPING_NODE).
+	mapping struct {
+		pairs_data  []NodePair   // The stack of mapping pairs (key, value).
+		pairs_start *NodePair    // The beginning of the stack.
+		pairs_end   *NodePair    // The end of the stack.
+		pairs_top   *NodePair    // The top of the stack.
+		style       MappingStyle // The mapping style.
+	}
+
+	start_mark Mark // The beginning of the node.
+	end_mark   Mark // The end of the node.
+}
+
+// The document structure.
+type Document struct {
+	// The document nodes.
+	nodes []Node
+
+	// The version directive.
+	version_directive *VersionDirective
+
+	// The list of tag directives.
+	tag_directives_data  []TagDirective
+	tag_directives_start int // The beginning of the tag directives list.
+	tag_directives_end   int // The end of the tag directives list.
+
+	start_implicit int // Is the document start indicator implicit?
+	end_implicit   int // Is the document end indicator implicit?
+
+	// The start/end of the document.
+	start_mark, end_mark Mark
+}
+
+// The prototype of a read handler.
+//
+// The read handler is called when the parser needs to read more bytes from the
+// source. The handler should write not more than size bytes to the buffer.
+// The number of written bytes should be set to the size_read variable.
+//
+// [in,out]   data        A pointer to an application data specified by
+//
+//	yamlParser.setInput().
+//
+// [out]      buffer      The buffer to write the data from the source.
+// [in]       size        The size of the buffer.
+// [out]      size_read   The actual number of bytes read from the source.
+//
+// On success, the handler should return 1.  If the handler failed,
+// the returned value should be 0. On EOF, the handler should set the
+// size_read to 0 and return 1.
+type ReadHandler func(parser *Parser, buffer []byte) (n int, err error)
+
+// This structure holds information about a potential simple key.
+type SimpleKey struct {
```

Comment:
@ccoVeille I think we shouldn't make all structs public, only the structs that we use outside of this package.

Reply:
You are right.

The risk is limited because everything is now in internal packages, so it's not exported outside the module.

I might need to double check, maybe the SimpleKey is now available via an exported field that become exported because of the changes.

I can fix that if needed.

Maybe it was a simple search and replace that went too far.

Thanks for catching this